### DISCUSSION TITLE
Poster cropped for certain model sizes/views

### DISCUSSION THREAD

**Discussion by axbylabs** [google/model-viewer#4894] [General]
Not sure if this is expected behaviour or not, but when a model is wider than it is taller, and is zoomed past a certain level then the poster generated by the Editor gets cropped.

For example, if you load the Macbeth sample model into the Editor and zoom in so that the model fills most of the screen, then generate a poster you will see that it is cropped at the sides.

It looks like this is due to the `useIdealAspect` setting where it calculates an x offset, but I'm not sure if there's a bug here or if this is a known limitation:
https://github.com/google/model-viewer/blob/3ed05fa99e8f0051caabd32d7e4527f154d0500e/packages/model-viewer/src/model-viewer-base.ts#L440

**Comment by elalish** [comment]
I would call this limitations of the editor - it could certainly be further improved, but it's pretty low priority, so PRs welcome! `useIdealAspect` is nice because it makes the poster seamless when the aspect ratio of the MV element is unknown (which is generally true for the editor). However, it doesn't handle zoom as nicely. It's pretty easy to set up your own screen-shotting page (you can even do it automatically using puppeteer) and call `toBlob` with your own parameters to get exactly what you need. This is what we assume everyone will do who need posters at scale.

**Reply by axbylabs** [reply]
Thanks for the response, yes it's seamless posters I'm looking to create, I'll see if I can code something to handle this scenario
